test/e2e/cases/hcFromProbe.go

```diff
@@ -0,0 +1,101 @@
+package e2e
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	kubeclient "k8s.io/client-go/kubernetes"
+
+	"github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/kubemci/pkg/gcp/cloudinterface"
+	utilsnamer "github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/kubemci/pkg/gcp/namer"
+)
+
+const (
+	lbNamePrefix = "mci1" // TODO get from loadbalancersyncer.go
+	svcNodePort = 30061
+	expectedHcPath = "/healthz"
+)
+
+func RunHCFromProbeTest() {
+	project, kubeConfigPath, lbName, ipName, clients := setupBasicDependencies()
+	setupHCFromProbe(kubeConfigPath, ipName, clients)
+	defer func() {
+		cleanupHCFromProbe(kubeConfigPath, ipName, clients)
+		cleanupBasicDependencies(ipName)
+	}()
+
+	testHCFromProbe(project, kubeConfigPath, lbName)
+}
+
+func setupHCFromProbe(kubeConfigPath, ipName string, clients map[string]kubeclient.Interface) {
```

Comment:
This can be a generic func that takes as input filepath for app spec (testdata/e2e/hcFromProbe/app/) and IP env var (E2E_KUBEMCI_IP) and then this and setupBasic can reuse that util func.